### DISCUSSION TITLE
How to secure useState's state?

### DISCUSSION THREAD

**Discussion by ariefsn** [nuxt/nuxt#18260] [Questions]
Hi,
wanna ask about `useState` [https://v3.nuxtjs.org/docs/usage/state](https://v3.nuxtjs.org/docs/usage/state)
I have created plugin like the example, and I give a key with `testGlobalState`

```
// store.server.ts
export default defineNuxtPlugin(nuxtApp => {
  const testGlobalState = useState('testGlobalState', () => 'This is global state')
})
```

![image](https://user-images.githubusercontent.com/33569737/138563348-c0153711-fe93-4508-ba63-5d4c463e87b4.png)

it's works properly.
but, it can be modified from browser console.

![image](https://user-images.githubusercontent.com/33569737/138563511-5ae24f9e-ed9f-482b-85f3-a405a550d92d.png)

how to avoid this?

thanks.

**Comment by wobsoriano** [comment]
Just like vuex stores - you can mutate it in browser console. :) As long as there's no sensitive data there it should be good.

**Reply by ariefsn** [reply]
I don't know if vuex can be mutated too, can you share how to do it?
at least, the state from vuex not showing in the inspect elements (I guess).
I'm just worried that it might be like the case I want to post the data as a payload, and it can be mutated from the console before I send the request.

**Reply by rslonik** [reply]
The user will always be able to change requests client side. Vuex state is available for the client too. So it's your server's responsible to deal with validation and stuff like that.
